### PR TITLE
Distribution csv sync

### PR DESCRIPTION
**Pull request by zashton**
Fixes
https://www.notion.so/coordinape/VAULTS-QA-dd5a668750314cbc8f9a431fac1132a0?p=5658986b4684436a8efe22b7eb4588d7&pm=s

Syncing the CSV with the allocation to show preview values

### CODE REVIEW

File: api/hasura/actions/_handlers/allocationCsv.ts [hunk 1]
```diff
@@ -4,7 +4,8 @@ import type { VercelRequest, VercelResponse } from '@vercel/node';
 import { v4 as uuidv4 } from 'uuid';
```
Comment (levity) [inline api/hasura/actions/_handlers/allocationCsv.ts:4]: @zashton can you please write a jest test for this, to verify that when you add certain inputs, they show up in the CSV? you can see an example of how to mock the request & response objects as well as the hasura query [here](https://github.com/coordinape/coordinape/blob/main/api/hasura/cron/_recoverVaults.test.ts) if needed. thanks!

Comment (levity) [reply api/hasura/actions/_handlers/allocationCsv.ts:4]: it may make sense to have a few different tests to cover the different cases (gift, fixed, combined)

File: api/hasura/actions/_handlers/allocationCsv.ts [hunk 2]
```diff
@@ -29,88 +32,120 @@ async function handler(req: VercelRequest, res: VercelResponse) {
...
+    d =>
+      d.distribution_type === DISTRIBUTION_TYPE.FIXED ||
+      d.distribution_type === DISTRIBUTION_TYPE.COMBINED
+  );
+
+  giftTokenSymbol = circleDist ? circleDist.vault.symbol : giftTokenSymbol;
+  const { users } = circle;
+
+  const userValues: (string | number)[][] = [];
   users.map((u, idx) => {
```
Comment (topocount) [inline api/hasura/actions/_handlers/allocationCsv.ts:125]: this should probably be `.forEach` instead of `.map` since we're not using the returned result.

File: api/hasura/actions/_handlers/allocationCsv.ts [hunk 3]
```diff
@@ -29,88 +32,120 @@ async function handler(req: VercelRequest, res: VercelResponse) {
...
     'sent',
-    'epoch_number',
-    'Date',
+    'givers',
+    'percentage_of_give',
+    'circle_rewards',
+    'circle_rewards_token',
   ];
   if (fixedPaymentsEnabled) {
     headers.push('fixed_payment_amount');
```
Comment (topocount) [inline api/hasura/actions/_handlers/allocationCsv.ts:73]:
```suggestion
    headers.push('fixed_payment_rewards');
```

File: api/hasura/actions/_handlers/allocationCsv.ts [hunk 4]
```diff
@@ -29,88 +32,120 @@ async function handler(req: VercelRequest, res: VercelResponse) {
...
+  const fixedPaymentsEnabled =
+    isFeatureEnabled('fixed_payments') && !!circle.fixed_payment_token_type;
+
+  const userValues = generateCsvValues(
+    circle,
+    form_gift_amount,
+    gift_token_symbol,
+    totalTokensSent,
+    fixedPaymentsEnabled,
+    circle.fixed_payment_token_type,
```
Comment (topocount) [inline api/hasura/actions/_handlers/allocationCsv.ts:57]: This doesnt seem like the best way to provide this value (from the circle details) since it can be changed by the user at any time. Why not use the same token if its a combined distribution, or use the vault symbol from the fixed distribution if its a separate distribution?

**Review by topocount — CHANGES_REQUESTED** [review]
I appreciate breaking this down into functions but it's still hard to follow everything that's going on. Maybe adding some comments here and there could help.

File: api/hasura/actions/_handlers/allocationCsv.ts (hunk 4 above)
Comment (levity) [reply api/hasura/actions/_handlers/allocationCsv.ts:57]: i agree this is a real issue, and the root cause is that we don't take a snapshot of the circle's settings at the end of the epoch. 

> Why not use the same token if its a combined distribution, or use the vault symbol from the fixed distribution if its a separate distribution?

we can't assume there has been a distribution.
